TodoApp/test/TodoApp.Services.Tests/Services/RetrievedItemTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Threading.Tasks;
+using NSubstitute;
+using NUnit.Framework;
+using TodoApp.Contracts.Models;
+using TodoApp.Contracts.Repositories;
+using TodoApp.Contracts.Services;
+using TodoApp.Services.Services;
+
+namespace TodoApp.Services.Tests.Services
+{
+    [TestFixture]
+    internal class RetrievedItemTests
+    {
+        private IItemRepository _repository;
+        private IGetItemByIdService _getItemByIdService;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _repository = Substitute.For<IItemRepository>();
+
+            _getItemByIdService = new GetItemByIdService(_repository);
+        }
+
+        [Test]
+        public async Task GetItemByIdAsync_ExistingId_ReturnsRetrievedItemWithSuccess()
```

Comment:
Tests in this class should not test `_getItemByIdService` at all. They should only test `RetreivedItem` itself - that is their unit. You basically test that not-null `Item` has `WasFound` set to `true` and `null` `Item` has `WasFound` set to `false` and trying to get `Item` for `WasFound` being `false` throws an exception.